### PR TITLE
Backport PR #14412 on branch 3.6.x (Right-align notification buttons, reduce outer padding, add space between buttons and message)

### PR DESCRIPTION
**Pull request by andrii-i**
## References

- Backports #14412 to 3.6.x
- Fixes visual glitch due to which both round and rectangular hover effects appeared on hover over announcement popup 

Closes #14503 

Without:
Without

With:
After

### CODE REVIEW

**Comment by andrii-i** [comment]
@fcollonval thank you for updating the name to more clear one! This PR needs a bit more work as closing buttons look "hovered over" by default. 

Screenshot 2023-05-08 at 6 25 21 PM

File: packages/apputils-extension/src/notificationplugin.tsx [hunk 1]
```diff
@@ -639,19 +641,53 @@ namespace Private {
...
+    closeIconMargin?: boolean;
+  }
+
+  export function CloseButton(props: ICloseButtonProps) {
     return (
       <button
-        className={`bp3-button bp3-minimal jp-Button jp-mod-minimal ${TOAST_CLOSE_BUTTON_CLASS}`}
-        title={trans.__('Hide notification')}
-        onClick={props.closeToast}
+        className={`jp-Button jp-mod-minimal ${TOAST_CLOSE_BUTTON_CLASS}${
```
Comment (fcollonval) [inline packages/apputils-extension/src/notificationplugin.tsx:669]: You need to keep the bp3 classes.

Comment (andrii-i) [reply packages/apputils-extension/src/notificationplugin.tsx]: Kept bp3 classes, removed "jp-icon-hover" from props.closeIcon that made both circular and rectangular hover effects appear, see PR description for updated preview

**Comment by andrii-i** [comment]
Bot please update galata snapshots

**Comment by andrii-i** [comment]
Kicking CI

**Comment by andrii-i** [comment]
Bot please update galata snapshots (again)

**Comment by andrii-i** [comment]
Kicking CI

**Review by fcollonval — APPROVED** [review]
Thanks @andrii-i

**Comment by andrii-i** [comment]
Thank you @fcollonval

**Pull request merged**